code/modules/vehicles/mecha/combat/greyscale/greyscale_constructor.dm

```diff
@@ -289,6 +306,11 @@ GLOBAL_LIST_INIT(greyscale_weapons_data, generate_greyscale_weapons_data())
 			if(isnull(selected_visor))
 				return FALSE
 			currently_assembling = TRUE
+			in_cooldown = TRUE
+			update_icon()
+			spawn(30 MINUTES)
+				in_cooldown = FALSE
+				update_icon()
```

Comment:
Spawn use is banned except in specific circumstances
Use the timer cooldown system

Reply:
my bad i didnt know that